ytypes/validate.go

```diff
@@ -22,9 +22,28 @@ import (
 	"github.com/openconfig/ygot/ygot"
 )
 
+// LeafrefOptions controls the behaviour of validation functions for leaf-ref
+// data types.
+type LeafrefOptions struct {
+	// IgnoreMissingData determines whether leafrefs that target a node
+	// that does not exist should return an error to the calling application. When
+	// set to true, no error is returned.
+	//
+	// This functionality is typically used where a partial set of schema information
+	// is populated, but validation is required - for example, configuration for
+	// a protocol within OpenConfig references an interface, but the schema being
+	// validated does not contain the interface definitions.
```

Comment:
Suggest a change to the wording to something like "that target a node that either doesn't exist or is not in the subtree of the element being validated"
Maybe mention that this could also be used in the case of validating a subtree rather than the root. In that case it would probably always be appropriate since leafrefs often require the tree root to navigate to.

Reply:
Per the analysis below, I'm not sure that this "is not in the subtree" is actually an error case that is ignored here? Can you see if you agree.